src/server/extractors.rs

```diff
@@ -0,0 +1,35 @@
+use regex::Regex;
+use actix_web::{
+    Error, FromRequest,
+};
+use futures::future::{ok, err, Ready};
+use actix_web::error::ErrorBadRequest;
+
+pub struct  ClientState {
+   extractedClientState: String,
```

Comment:
The end result we need is just one extracted and validated String, so you can get rid of `validatedClientState`.

Be sure to check the rustc compiler warnings, it prefers snake cased names (granted this is a WIP):

```
warning: structure field `extractedClientState` should have a snake case name
 --> src/server/extractors.rs:9:4
  |
9 |    extractedClientState: String,
  |    ^^^^^^^^^^^^^^^^^^^^ help: convert the identifier to snake case: `extracted_client_state`
```

I'd call this one simply "client_state" or "value": the "extracted" quality is really implied by anything implementing `FromRequest`.

Reply:
Ok. I have changed the names to snake case.